.coveragerc

```diff
@@ -642,6 +642,7 @@ omit =
     homeassistant/components/linux_battery/sensor.py
     homeassistant/components/lirc/*
     homeassistant/components/livisi/__init__.py
+    homeassistant/components/livisi/binary_sensor.py
```

Comment:
With all the refactoring going on, I think you should consider adding basic tests.

With the correct fixtures it can be as simple as:
```python
@patch("homeassistant.components.livisi.PLATFORMS", [Platform.BINARY_SENSOR])
async def test_binary_sensors(
    hass: HomeAssistant,
    config_entry: ConfigEntry,
    device_registry: dr.DeviceRegistry,
    entity_registry: er.EntityRegistry,
    snapshot: SnapshotAssertion
) -> None:
    """Test for binary sensors."""
    await hass.config_entries.async_setup(config_entry.entry_id)
    await hass.async_block_till_done()

    device_entries = dr.async_entries_for_config_entry(
        device_registry, config_entry.entry_id
    )
    assert device_entries == snapshot

    entity_entries = er.async_entries_for_config_entry(
        entity_registry, config_entry.entry_id
    )
    assert entity_entries == snapshot

    assert hass.states.async_all(Platform.BINARY_SENSOR) == snapshot
```

Reply:
I will look into this, thanks for the example

Reply:
Looks like this needs a lot more effort for me to understand what's going on in that snippet and how to do the testing right.

I'd rather have the WDS support merged soon (the original PR was from december) and look at testing (for all the existing devices) later in a a separate PR. I understand you are quite picky about code quality and normally I see this a a good thing, but the livisi implementation is still very incomplete and not very useful in its current state, so it would be great to see some progress in supported devices.

You may have noticed I'm very open to refactoring code for simplification and unification - so I will definitely continue to work on improving the code quality in the future.

Reply:
It's OK to keep tests for a follow-up PR... it's just something to keep in mind.